html/modules/custom/docstore/src/Commands/DocstoreCommands.php

```diff
@@ -0,0 +1,459 @@
+
+  // Drush traits.
+  use ProcessManagerAwareTrait;
+  use SiteAliasManagerAwareTrait;
+
+  // Docstore traits.
+  use FileTrait;
+  use ResourceTrait;
+
+  /**
+   * The current user service.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * The entity field manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * The entity manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityRepositoryInterface
+   */
+  protected $entityRepository;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The mime type guesser service.
+   *
+   * @var \Drupal\Core\ProxyClass\File\MimeType\MimeTypeGuesser
+   */
+  protected $mimeTypeGuesser;
+
+  /**
+   * The file system.
+   *
+   * @var \Drupal\Core\File\FileSystem
+   */
+  protected $fileSystem;
+
+  /**
+   * The file usage.
+   *
+   * @var \Drupal\file\FileUsage\FileUsageInterface
+   */
+  protected $fileUsage;
+
+  /**
+   * The state store.
+   *
+   * @var \Drupal\Core\State\State
+   */
+  protected $state;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+      AccountInterface $current_user,
+      ConfigFactoryInterface $config_factory,
+      Connection $database,
+      EntityFieldManagerInterface $entity_field_manager,
+      EntityRepositoryInterface $entity_repository,
+      EntityTypeManagerInterface $entity_type_manager,
+      MimeTypeGuesser $mimeTypeGuesser,
+      FileSystem $file_system,
+      FileUsageInterface $file_usage,
+      State $state
+    ) {
+    $this->currentUser = $current_user;
+    $this->configFactory = $config_factory;
+    $this->database = $database;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->entityRepository = $entity_repository;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->mimeTypeGuesser = $mimeTypeGuesser;
+    $this->fileSystem = $file_system;
+    $this->fileUsage = $file_usage;
+    $this->state = $state;
+  }
+
+  /**
+   * Reset the docstore for testing.
+   *
+   * @command docstore:test-reset
+   * @usage docstore:test-reset
+   *   Reset the docstore for testing.
+   * @validate-module-enabled docstore
+   */
+  public function resetTesting() {
+    // Create/reset the test users.
+    $this->createTestUsers();
+
+    // Reload the configuration of the solr indices.
+    // @todo do we really need to do that? This looks like the site's config
+    // may become out of sync if those files are not properly maintained.
```

Comment:
Site config will be out of date for entities and fields, since those are created using the API

Reply:
Thanks. I need to remove this todo, it was before I understood how the solr config was generated.